src/pds/cross_report.rs

```diff
@@ -0,0 +1,404 @@
+            let filter_status = filter_storage
+                .try_consume(&filter_id, &individual_privacy_loss)?;
+
+            if filter_status == FilterStatus::OutOfBudget {
+                // Not enough budget, drop events without any filter
+                // consumption
+                relevant_events.drop_epoch(&epoch_id);
+
+                // Keep track of why we dropped this epoch
+                oob_filters.push(filter_id);
+            }
+        }
+
+        // Now that we've dropped OOB epochs, we can compute the final report.
+        let mut filtered_report = self.request.compute_report(&relevant_events);
+
+        // only keep the buckets that are requested
+        filtered_report.bin_values.retain(|bucket, _| {
+            relevant_event_selector.requested_buckets.contains(bucket)
+        });
+
+        let report = PdsReport {
+            filtered_report,
+            unfiltered_report,
+            oob_filters,
+        };
+        Ok(report)
+    }
+}
+
+impl<U, FS, ERR> PrivateDataServiceCore<PpaHistogramRequest<U>, FS, ERR>
+where
+    U: Uri,
+    FS: FilterStorage<
+        FilterId = FilterId<PpaEpochId, U>,
+        Budget = PureDPBudget,
+    >,
+    ERR: From<FS::Error>,
+{
+    pub fn measure_conversion(
+        &mut self,
+        request: PpaHistogramRequest<U>,
+        mut relevant_events: RelevantEvents<PpaEvent<U>>,
+    ) -> Result<AttributionObject<PpaHistogramRequest<U>>, ERR> {
+        let uris = request.report_uris();
+
+        let epochs = request.epoch_ids();
+        let num_epochs = epochs.len();
+
+        // Compute the raw report, useful for debugging and accounting.
+        let unfiltered_result = request.compute_report(&relevant_events);
+
+        let mut oob_filters = vec![];
+        for epoch_id in epochs {
+            let individual_privacy_loss =
+                request.histogram_multi_epoch_report_global_sensitivity();
+
+            let source_losses = compute_epoch_source_losses(
+                &request,
+                relevant_events.sources_for_epoch(&epoch_id),
+                &unfiltered_result,
+                num_epochs,
+            );
+
+            // Try to consume budget from current epoch, drop events if OOB.
+            // Two phase commit.
+            let mut filters_to_consume = self.filters_to_consume(
+                epoch_id,
+                &individual_privacy_loss,
+                &source_losses,
+                request.report_uris(),
+            );
+
+            // Do not consume per-querier, that is done in get_report().
+            // Change the per-querier budgets to 0 to still initialize the
+            // filter.
+            for querier_uri in &uris.querier_uris {
+                filters_to_consume.insert(
+                    FilterId::PerQuerier(epoch_id, querier_uri.clone()),
+                    &0.0, // no budget consumption for per-querier filter
+                );
+            }
```

Comment:
What happens if you drop this and don't initialize the filters?

Reply:
Nothing if we assume static capacities. But if the default per-site capacity changes between now and `get_report()`, then I'm guessing the privacy proofs don't apply anymore

Reply:
Changed `.insert(0.0)` to `.remove()` in https://github.com/columbia/pdslib/pull/95/commits/614c352e9e34d709433c1c2c40cd3e4886599477